IronFoundry.Container.Shared/Utilities/LocalPrincipalManager.cs

```diff
@@ -162,5 +164,29 @@ private void AddUserToGroup(PrincipalContext context, string groupName, UserPrin
 
             group.Save();
         }
+
+        private Principal InvokeSearcherWithRetries(PrincipalSearcher searcher)
+        {
+            Principal group = null;
+
+            Func<bool> findGroup = () =>
+            {
+                try
+                {
+                    group = searcher.FindOne() as GroupPrincipal;
+                }
+                catch (COMException)
```

Comment:
Consider checking for the error code here (i.e. `COM_EXCEPT_UNKNOWN_DIRECTORY_OBJECT: 0x80005004`).